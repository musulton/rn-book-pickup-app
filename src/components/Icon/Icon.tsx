import Ionicons from "@expo/vector-icons/Ionicons";

import React from "react";

interface IconProps {
    name: keyof typeof Ionicons.glyphMap
    size: number
    color?: string
    onPress?: () => void
}

const Icon: React.FC<IconProps> = (props) => {
    return (
        <Ionicons
            name={props.name}
            size={props.size}
            color={props.color}
            onPress={props.onPress}
        />
    )
}

export default Icon
