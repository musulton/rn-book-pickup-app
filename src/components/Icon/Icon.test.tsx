import Ionicons from "@expo/vector-icons/Ionicons";
import {render, screen} from "@testing-library/react-native";

import Icon from "./Icon";

describe("<Icon />", () => {
    it("should render correctly", () => {
        const props = {
            name: "close" as keyof typeof Ionicons.glyphMap,
            color: "white",
            size: 30,
            onPress: jest.fn()
        }

        render(<Icon {...props} />)

        expect(screen.toJSON()).toMatchSnapshot();
    })
});
