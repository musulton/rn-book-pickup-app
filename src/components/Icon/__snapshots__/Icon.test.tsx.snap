// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> should render correctly 1`] = `<Text />`;
